sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -565,6 +565,7 @@ abstract class HadoopFsRelation private[sql](maybePartitionSpec: Option[Partitio
       filters: Array[Filter],
       inputPaths: Array[String],
       broadcastedConf: Broadcast[SerializableConfiguration]): RDD[Row] = {
+    refresh()
```

Comment:
Yeah, I'll probably work on this later this week, it can be relatively tricky to handle...